internal/grpc/commander/channel.go

```diff
@@ -0,0 +1,122 @@
+package commander
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/nginx/agent/sdk/v2/proto"
+	"golang.org/x/sync/errgroup"
+)
+
+const channelLength = 25
+
+// BidirectionalChannel encapsulates the CommandChannelServer which is a bidirectional streaming channel.
+// The BidirectionalChannel is responsible for sending and receiving commands to and from the CommandChannelServer.
+//
+// All commands received from the CommandChannelServer are put on the fromClient channel and can be accessed through
+// the Out() method.
+//
+// Commands can be sent to the CommandChannelServer by placing them on the toClient channel,
+// which is accessible through the In() method.
+//
+// To use the BidirectionalChannel you must call the Run() method to kick of the receive and send loops.
```

Comment:
What would happen if the client accessed those channels before run()?  I think in the worst case the client would block on the In() channel after placing 25 commands on it. Once the channel started running it would unblock the client, no? Is there an unrecoverable error I'm missing?

Reply:
Spoke offline. For now, we will leave this as is, but I will look into possible solutions in the next agent PR.